src/schemes/boundary/open_boundary/method_of_characteristics.jl

```diff
@@ -98,9 +102,16 @@ function evaluate_characteristics!(system, v, u, v_ode, u_ode, semi, t)
                 end
             end
 
-            characteristics[1, particle] = avg_J1 / counter
-            characteristics[2, particle] = avg_J2 / counter
-            characteristics[3, particle] = avg_J3 / counter
+            # To prevent NANs here if the boundary has not been in contact before.
+            if counter > 0
+                characteristics[1, particle] = avg_J1 / counter
+                characteristics[2, particle] = avg_J2 / counter
+                characteristics[3, particle] = avg_J3 / counter
+            else
+                characteristics[1, particle] = 0.0
+                characteristics[2, particle] = 0.0
+                characteristics[3, particle] = 0.0
```

Comment:
```suggestion
                characteristics[1, particle] = 0
                characteristics[2, particle] = 0
                characteristics[3, particle] = 0
```
Should be fine here, but let's avoid `Float64` literals.